lib/reek/smells/smell_detector.rb

```diff
@@ -60,41 +38,20 @@ def smell_type
         self.class.smell_type
       end
 
-      class << self
```

Comment:
We opened up "self" for the second time here which wasn't only unnecessary but also confusing and bad for readability.